x-pack/plugins/fleet/server/constants/fleet_es_assets.ts

```diff
@@ -17,6 +17,8 @@ export const FLEET_AGENT_POLICIES_SCHEMA_VERSION = '1.1.1';
 
 export const FLEET_FINAL_PIPELINE_ID = '.fleet_final_pipeline-1';
 
+export const FLEET_ECS_MAPPINGS_COMPONENT_TEMPLATE_NAME = 'ecs@mappings';
```

Comment:
What will create that component template? it is built in in ES?

Reply:
it is built-in in ES. Link https://github.com/elastic/elasticsearch/blob/4717f9c5c12dd12c4bb542996bdd728b7d0e9a02/x-pack/plugin/core/template-resources/src/main/resources/ecs%40mappings.json.

I'm currently debugging an issue with a datastream of type metric `nginx.stubstatus` (that I used just for testing). 

This error only happens with metric datastream

```
[2024-01-15T10:28:13.125+00:00][WARN ][plugins.fleet] Failure to install package [nginx]: [ResponseError: mapper_parsing_exception
        Root causes:
                mapper_parsing_exception: no type specified for property [text]]
[2024-01-15T10:28:13.126+00:00][ERROR][plugins.fleet] uninstalling nginx-1.17.0 after error installing: [ResponseError: mapper_parsing_exception
        Root causes:
                mapper_parsing_exception: no type specified for property [text]]
[2024-01-15T10:30:56.445+00:00][ERROR][plugins.fleet] FleetError: Error deleting component template ecs@mappings
```

What troubles me more than the mapping exception is the fact that at any error, it will try to remove the component template. Since it is a built-in component template, it cannot be removed and fails even harder.

Reply:
the relevant code here https://github.com/nchaulet/kibana/blob/10e02addcf80127ef07c1ceb2969feda8674af0d/x-pack/plugins/fleet/server/services/epm/packages/remove.ts#L166

Reply:
the issue with the mapper_parsing_exception has now been fixed at https://github.com/elastic/elasticsearch/pull/104418

Reply:
yes, should we make it part of this PR? @felixbarny

Reply:
Fleet should definitely not remove `ecs@mappings`. There are also other component templates that are imported by fleet that Elasticsearch sets up (see https://github.com/elastic/kibana/issues/163141). Fleet must never uninstall them.